h2o-core/src/main/java/water/util/MRUtils.java

```diff
@@ -282,8 +286,14 @@ public void map(Chunk[] cs, NewChunk[] ncs) {
             sampling_reps = (int)sampling_ratios[label] + (rng.nextFloat() < remainder ? 1 : 0);
           }
           for (int i = 0; i < ncs.length; i++) {
-            for (int j = 0; j < sampling_reps; ++j) {
-              ncs[i].addNum(cs[i].atd(r));
+            if (cs[i] instanceof CStrChunk) {
```

Comment:
I know this is not ideal, but better than before, and good enough to work for most datasets (those without UUID cols).